src/Abp.ZeroCore/Authorization/Roles/AbpRoleManager.cs

```diff
@@ -178,17 +178,10 @@ public virtual async Task<IReadOnlyList<Permission>> GetGrantedPermissionsAsync(
         /// <returns>List of granted permissions</returns>
         public virtual async Task<IReadOnlyList<Permission>> GetGrantedPermissionsAsync(TRole role)
         {
-            var permissionList = new List<Permission>();
-
-            foreach (var permission in _permissionManager.GetAllPermissions())
-            {
-                if (await IsGrantedAsync(role.Id, permission))
-                {
-                    permissionList.Add(permission);
-                }
-            }
-
-            return permissionList;
+            var cacheItem = await GetRolePermissionCacheItemAsync(role.Id);
+            var allPermissions = _permissionManager.GetAllPermissions();
+            var grantedPermissions = allPermissions.Select(x => x.Name).Intersect(cacheItem.GrantedPermissions);
```

Comment:
No need to intersect since we already filter in the next line?

```suggestion
            var grantedPermissions = cacheItem.GrantedPermissions;
```

Reply:
Yes, I forgot this.